one-off/nuke-workitem-embedded-images.sql

```diff
@@ -0,0 +1,174 @@
+            charindex('"', @richtextcol, charindex('downloadblob.img/', @richtextcol, startpos) + len('downloadblob.img/')) - charindex('downloadblob.img/', @richtextcol, startpos) - len('downloadblob.img/')
+        ),1) as hash
+    from
+        recursivecte
+    where
+        charindex('downloadblob.img/', @richtextcol, startpos) > 0
+)
+
+GO
+
+
+create or alter function [dbo].[ReplaceBase64Images] (@RichTextCol varchar(max), @BlankBase64Image varchar(max))
+returns varchar(max)
+as
+begin
+    declare @StartIndex int = charindex('<img src="data:image/', @RichTextCol);
+
+    while @StartIndex > 0
+    begin
+        set @RichTextCol = stuff(
+            @RichTextCol,
+            @StartIndex,
+            charindex('">', substring(@RichTextCol, @StartIndex, len(@RichTextCol))) + 2,
+            @BlankBase64Image
+        );
+
+        set @StartIndex = charindex('<img src="data:image/', @RichTextCol, @StartIndex + len(@BlankBase64Image));
+    end;
+
+    return @RichTextCol;
+end;
+GO
+
+
+create or alter procedure dbo.NukeWorkitemEmbeddedImages
+    @oid varchar(20),
+	@saveChanges bit
+as
+begin
+    declare @Id int, @assetType varchar(100)
+
+	declare @error int, @rowcount int
+
+    select @Id = ba.ID, @assetType = ba.AssetType
+    from BaseAsset_Now ba
+	join dbo.OidSplit(@oid) oid on ba.ID = oid.ID and ba.AssetType = oid.AssetType
+
+    if (@@rowcount <> 1)
+    begin
+        raiserror('Invalid asset oid', 0, 1)
+        return
+    end
+
+    declare @blankimage varbinary(max) = 0x89504e470d0a1a0a0000000d4948445200000001000000010100000000376ef9240000001049444154789c626001000000ffff03000006000557bfabd40000000049454e44ae426082
+	declare @blankImageBase64 varchar(max) = CAST(N'' AS XML).value('xs:base64Binary(xs:hexBinary(sql:variable("@blankimage")))', 'VARCHAR(MAX)')
+    declare @imgpng int
+    
+    begin tran; save tran TX
+
+    exec dbo._SaveString N'img/png', @imgpng output
+
+    update Blob 
+    set Value = @blankimage, ContentType=@imgpng
+    where ID in (
+        select Content
+        from dbo.EmbeddedImage
+        where AssetID = @Id
+    )
+    and cast(Value as varbinary(max)) <> @blankimage
+
+	select @rowcount=@@ROWCOUNT, @error=@@ERROR
+	if @error<>0 goto ERR
+    raiserror('%d Embedded Images nuked', 0, 1, @rowcount) with nowait
+
+
+	DECLARE cRichTextFields CURSOR FORWARD_ONLY FOR
+	select ls.ID, cast (ls.Value as nvarchar(max))
+	from BaseAsset bs
+	left join CustomLongText clt on bs.ID = clt.ID
+	left join LongString ls on ls.ID in (bs.Description, clt.Value)
+	where bs.ID = @Id and ls.ID is not null
+
+	OPEN cRichTextFields;
+
+	DECLARE @LongStringID int, @Value nvarchar(max)
+
+	FETCH NEXT FROM cRichTextFields INTO @LongStringID, @Value
+
+	WHILE @@FETCH_STATUS = 0
+	BEGIN
+		update Blob 
+		set Value = @blankimage, ContentType=@imgpng
+		where Hash in (
+			SELECT Hash
+			FROM dbo.ExtractHashesFromUrls(@Value)
+		)
```

Comment:
welp, `Blob.Hash` is now a lie

i think we need to
1. create a new `Blob` with `@blankimage` value and matching (pre-computed) hash
2. delete every `Blob` with doomed `Hash` from the extracted list
3. rewrite original `LongString.Value`, replacing doomed hashes with the known blank image hash

Reply:
> don't think Blob.Hash needs to actually correlate to the content.

true, it does not _currently_, as the hash value is being treated as an opaque unique blob ID

> I have a branch in progress somewhere that replaces it with an random value

i know we discussed it before on another PR somewhere. i'd prefer for `Hash` to stay as a true hash, if (when?) we ever decide to de-dupe the `Blob` values like we now do for `LongString`

Reply:
> We could dedupe using another true hash, maybe an indexed calculated column, calculated db-side

why, though? existing value already _is_ a true `sha1` hash of the data

> clients that use the current `Hash` like an ID have to follow an upload with a query to fetch it

not sure what you mean, the hash is computed on-the-fly during blob upload, and its hashed URL is returned in upload response